backends/tc/ebpfCodeGen.cpp

```diff
@@ -1235,7 +1235,10 @@ const PNAEbpfGenerator *ConvertToEbpfPNA::build(const IR::ToplevelBlock *tlb) {
             !d->is<IR::Type_Error>()) {
             if (d->srcInfo.isValid()) {
                 auto sourceFile = d->srcInfo.getSourceFile();
-                if (sourceFile.endsWith("/pna.p4")) {
+                // TODO: Similar logic exists in frontends/p4/toP4/toP4.cpp. Consider replacing
```

Comment:
It seems like the parser should already determine whether an object is source from a systems file and add this information to the source information.

Reply:
@ChrisDodd Do you think this could be done in the P4 parser? How much work would it be?